### PR TITLE
Codegen expressions that contain +/-/* operators on float8 data types

### PR DESCRIPTION
**Pull request by armenatzoglou**
This PR enhances the functionality of codegen's expression evaluation framework with support of  +/-/* operators on float8 data types. However, current solution does NOT support overflow.

@karthijrk @hardikar @foyzur Please take a look if you have some time. Thanks

### CODE REVIEW

File: src/backend/codegen/include/codegen/pg_arith_func_generator.h [hunk 1]
```diff
@@ -144,30 +151,36 @@ bool PGArithFuncGenerator<rtype, Arg0, Arg1>::AddWithOverflow(
...
+                          llvm_overflow_block,
+                          llvm_non_overflow_block );
+
+    irb->SetInsertPoint(llvm_overflow_block);
+    // TODO : krajaraman Elog::ERROR after ElogWrapper integrad.
+    irb->CreateBr(llvm_error_block);
+    irb->SetInsertPoint(llvm_non_overflow_block);
+  }
+  else {
+    *llvm_out_value = llvm_add_output;
```
Comment (karthijrk) [inline src/backend/codegen/include/codegen/pg_arith_func_generator.h]: will a+b+c work ? Do we need to do bitcast to int64 ?

**Comment by karthijrk** [comment]
LGTM 👍

File: src/backend/codegen/include/codegen/pg_arith_func_generator.h [hunk 2]
```diff
@@ -107,30 +107,37 @@ bool PGArithFuncGenerator<rtype, Arg0, Arg1>::MulWithOverflow(
...
-             "mul_overflow_block", llvm_main_func);
 
   llvm::Value* llvm_mul_output = codegen_utils->CreateMulOverflow<rtype>(
       llvm_args[0], llvm_args[1]);
 
   llvm::IRBuilder<>* irb = codegen_utils->ir_builder();
 
-  *llvm_out_value = irb->CreateExtractValue(llvm_mul_output, 0);
-  llvm::Value* llvm_overflow_flag = irb->CreateExtractValue(llvm_mul_output, 1);
+  if (!llvm_mul_output->getType()->isDoubleTy())
```
Comment (hardikar) [inline src/backend/codegen/include/codegen/pg_arith_func_generator.h]: I would rather we have a positive check. i.e we support int types only vs not float types.

File: src/backend/codegen/include/codegen/utils/codegen_utils.h [hunk 3]
```diff
@@ -1340,6 +1340,55 @@ class ArithOpMaker<float> {
...
+    return generator->ir_builder()->CreateFMul(casted_arg0, casted_arg1);
+  }
+
+
+ private:
+  static void Checker(llvm::Value* arg0,
+                      llvm::Value* arg1) {
+    assert(nullptr != arg0 && nullptr != arg0->getType());
+    assert(nullptr != arg1 && nullptr != arg1->getType());
+    assert(arg0->getType()->isIntegerTy());
```
Comment (hardikar) [inline src/backend/codegen/include/codegen/utils/codegen_utils.h]: Er?? isintegetTy for when we expect float? I think this check input method can be push to the generic implementation of the class.

**Comment by hardikar** [comment]
Rest LGTM.

File: src/backend/codegen/include/codegen/utils/codegen_utils.h (hunk 3 above)
Comment (hardikar) [inline src/backend/codegen/include/codegen/utils/codegen_utils.h]: Created a chore (Organize ArithOpMaker) to do that.

File: src/backend/codegen/include/codegen/pg_arith_func_generator.h (hunk 1 above)
Comment (hardikar) [inline src/backend/codegen/include/codegen/pg_arith_func_generator.h]: Looks like that works fine because sizeof(float8) == sizeof(Datum)

**Comment by hardikar** [comment]
Pushed 09f005812de6952430b886e0321768290ae68e8c

**Pull request closed without merging**